src/pkg/publisher/publisher.go

```diff
@@ -0,0 +1,102 @@
+package publisher
+
+import "log"
+
+type Publisher struct {
+	subscribers []Subscriber
```

Comment:
Maybe we can add into Subscriber interface new func like `Name() string` and after that, we will use here `map` instead of `slice`

Reply:
\+